agent/agent.go

```diff
@@ -297,3 +305,29 @@ func (a *Agent) killProcs(tc *taskContext) {
 	}
 	grip.Infof("processes cleaned up for task %s", tc.task.ID)
 }
+
+func (a *Agent) runSetupScript(ctx context.Context) (string, error) {
+	script := filepath.Join(a.opts.WorkingDirectory, hostinit.SetupScriptName)
+	if _, err := os.Stat(script); os.IsNotExist(err) {
+		return "", nil
+	}
+
+	var cancel context.CancelFunc
+	ctx, cancel = context.WithTimeout(ctx, setupTimeout)
+	defer cancel()
+	cmd := a.getShCommandWithSudo(ctx, script)
+	out, scriptErr := cmd.CombinedOutput()
```

Comment:
why do we have a new variable for this err? use a

Reply:
We want to hold onto this error to return it on 325. If we can't remove the script though, we will return on 322.

Reply:
why not just use a collector and return everything?

Reply:
done